flow360/component/flow360_params/unit_system.py

```diff
@@ -342,8 +351,16 @@ def validate(vec_cls, value):
                     isinstance(value, _Flow360BaseUnit) and isinstance(value.val, Collection)
                 )
 
-                if not is_collection or len(value) != 3:
-                    raise TypeError(f"arg '{value}' needs to be a collection of 3 values")
+                if length is None:
```

Comment:
I vaguely remember that this does not work if I give a scalar value/dict/set as argument. And I also vaguely remember that if I pass in a flow_360_unit_system value ((0,1,0)*flow_360_unit_system.length for example) this also breaks.

But I might be wrong and this has been fixed.

Reply:
I sent a commit showcasing the ones that does not work.